vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/EagerApplicationScopedExtension.java

```diff
@@ -0,0 +1,38 @@
+package br.com.caelum.vraptor.ioc.cdi.extension;
+
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.util.AnnotationLiteral;
+
+import br.com.caelum.vraptor.ioc.cdi.EagerInitialized;
+
+/**
+ * When enabled, it allows users to use {@link EagerInitialized} annotation 
+ * to get your application scoped beans initialized just after CDI deployment.
+ *
+ * @author Rodrigo Turini
+ */
+public class EagerApplicationScopedExtension implements Extension {
+	
+	@SuppressWarnings("serial")
+	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
+		
+		Set<Bean<?>> beans = beanManager.getBeans(Object.class, 
+				new AnnotationLiteral<EagerInitialized>() {});
+		
+		for (Bean<?> bean : beans) {
+			Class<?> beanClass = bean.getBeanClass();
+			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
+				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
+				beanManager.getReference(bean, beanClass, ctx).toString();
```

Comment:
Ok, then. We have different perspectives about this subject :). For me, the
happy path is to delete VRaptor's annotations in order to use annotations
from the specs.

Em Wed Feb 04 2015 at 10:56:30 AM, Rodrigo Turini <notifications@github.com>
escreveu:

> In
> vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/EagerApplicationScopedExtension.java
> <https://github.com/caelum/vraptor4/pull/937#discussion_r24080799>:
>
> > + *
> > + * @author Rodrigo Turini
> > + */
> > +public class EagerApplicationScopedExtension implements Extension {
> > +	
> > +	@SuppressWarnings("serial")
> > +	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
> > +		
> > +		Set<Bean<?>> beans = beanManager.getBeans(Object.class,
> > +				new AnnotationLiteral<EagerInitialized>() {});
> > +		
> > +		for (Bean<?> bean : beans) {
> > +			Class<?> beanClass = bean.getBeanClass();
> > +			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
> > +				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
> > +				beanManager.getReference(bean, beanClass, ctx).toString();
>
> the point is that using EJB annotations, just like @DependsOn, in an app
> server will
> work. That's not the case here. We'll provide a bunch of annotations that
> doesn't
> work for servlet container. The goal of this PR is help this kind of user.
> After all, we
> don't need this Extension for application server...
>
> Finally, I really believe that our focus should be the application servers
>
> Sure, but we shouldn't forget about non app server users... there's a lot
> of them =)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caelum/vraptor4/pull/937/files#r24080799>.
>